mayastor/src/bdev/nexus/nexus_bdev_children.rs

```diff
@@ -580,4 +401,170 @@ impl Nexus {
+
+        let source = match self
+            .children
+            .iter_mut()
+            .find(|c| c.state == ChildState::Open)
+        {
+            Some(child) => child.name.clone(),
+            None => {
+                return Err(Error::OpenChildNotFound {
+                    name: self.name.clone(),
+                })
+            }
+        };
+
+        if let Some(dst_child) =
+            self.children.iter_mut().find(|c| c.name == name)
+        {
+            let task = RebuildTask::create(
+                &self.name.clone(),
+                &source,
+                name,
+                self.data_ent_offset,
+                self.bdev.num_blocks() + self.data_ent_offset,
+                |nexus, task| {
+                    Reactors::current().send_future(async move {
+                        Nexus::complete_rebuild(nexus, task).await;
+                    });
+                },
+            )
+            .context(StartRebuild {
+                child: name.to_string(),
+                name: self.name.clone(),
+            })?;
+
+            dst_child.repairing = true;
+
+            Ok(task.start())
+        } else {
+            Err(Error::ChildNotFound {
+                name: self.name.clone(),
+                child: name.to_owned(),
+            })
+        }
+    }
+
+    /// Stop the child's rebuild task in the background
+    pub async fn stop_rebuild(&mut self, name: &str) -> Result<(), Error> {
+        let rt = self.get_rebuild_task(name)?;
+        rt.stop();
+        Ok(())
+    }
+
+    /// Return the state of a rebuild task
+    pub async fn get_rebuild_state(
+        &mut self,
+        name: &str,
+    ) -> Result<RebuildStateReply, Error> {
+        let rt = self.get_rebuild_task(name)?;
+        Ok(RebuildStateReply {
+            state: rt.state.to_string(),
+        })
+    }
+
+    /// Return rebuild task associated with the child name.
+    /// Return error if no rebuild task associated with it.
+    fn get_rebuild_task<'a>(
+        &mut self,
+        name: &'a str,
+    ) -> Result<&'a mut RebuildTask, Error> {
+        if let Ok(task) = RebuildTask::lookup(name) {
+            if task.nexus == self.name {
+                return Ok(task);
+            }
+        }
+
+        Err(Error::RebuildTaskNotFound {
+            child: name.to_owned(),
+            name: self.name.clone(),
+        })
+    }
+
+    fn get_child_by_name(
+        &mut self,
+        name: &str,
+    ) -> Result<&mut NexusChild, Error> {
+        match self.children.iter_mut().find(|c| c.name == name) {
+            Some(child) => Ok(child),
+            None => Err(Error::ChildNotFound {
+                child: name.to_owned(),
+                name: self.name.clone(),
+            }),
+        }
+    }
+
+    /// On rebuild task completion it updates the child state and removes the
+    /// rebuild task in case of failure the child is left in a Faulted State
```

Comment:
Copy paste error in comment. This function doesn't remove the rebuild task. That is performed by the below "on_rebuild_complete" function

Reply:
yeah and even the comment is making non sense to me now, I suspect the formatter joined two sentences